bundle/src/main/java/com/adobe/acs/commons/packaging/util/AssetPackageUtil.java

```diff
@@ -0,0 +1,242 @@
+    }
+
+    /**
+     * Generates the filter paths by recursing through the content and returns the list.
+     *
+     * @return the compiled list of {@link PathFilterSet} objects with assets and pages.
+     */
+    public List<PathFilterSet> getPackageFilterPaths() {
+        final String pagePath = properties.get(PN_PAGE_PATH, String.class);
+        final Set<PathFilterSet> packageFilterPathSet;
+        if (StringUtils.isBlank(pagePath)) {
+            packageFilterPathSet = Collections.emptySet();
+        } else {
+            packageFilterPathSet = this.findAssetPaths(resourceResolver, pagePath);
+        }
+
+        // Add the page path as a filter unless it is explicitly excluded
+        this.addPagePath(packageFilterPathSet, pagePath, properties.get(PN_EXCLUDE_PAGES, false));
+
+        List<PathFilterSet> packageFilterPaths = new ArrayList<>(packageFilterPathSet);
+        // Reverse the new list to put the page inclusion at the top
+        Collections.reverse(packageFilterPaths);
+        return packageFilterPaths;
+    }
+
+    /**
+     * Takes a list of Strings that should be valid patterns. If they are not then the exception is
+     * logged and the pattern is not added to the list of checked patterns.
+     *
+     * @param input The array of intended patterns
+     * @return The converted array of patterns
+     */
+    private List<Pattern> generatePatterns(String[] input) {
+        final List<Pattern> patterns = new ArrayList<>();
+        for (String item : input) {
+            try {
+                if (StringUtils.isBlank(item)) {
+                    continue;
+                }
+                patterns.add(Pattern.compile(item));
+            } catch (PatternSyntaxException e) {
+                log.error("Pattern invalid, skipping. Pattern value: " + item);
+            }
+        }
+        return patterns;
+    }
+
+    /**
+     * Recursively iterate over the parent path specified in the configuration to aggregate all the
+     * String or String Array property values that are referencing a path in the DAM.
+     *
+     * @param resourceResolver Resolver from the current Request
+     * @param pagePath         The page path
+     * @return The full list of filter sets
+     */
+    private Set<PathFilterSet> findAssetPaths(final ResourceResolver resourceResolver,
+                                               final String pagePath) {
+
+        final Set<PathFilterSet> filters = new LinkedHashSet<>();
+        final Resource parentResource = resourceResolver.resolve(pagePath);
+        if (isExcluded(this.pageExclusionPatterns, pagePath)) {
+            if (pagePath.endsWith("/" + NameConstants.NN_CONTENT)) {
+                return filters;
+            }
+            final PageManager pageManager = resourceResolver.adaptTo(PageManager.class);
+            Page page = pageManager.getContainingPage(parentResource);
+            excludedPages.add(page.getPath());
```

Comment:
A bit of an edge case, but when I test a page exclude path of `.*men.*` I get a NullPointerException on this line that blows up the package creation.

The reason is it finds `/content/we-retail/us/en/community/configuration/functions/assignments` where there is no "containing page".  The same error happens with full paths of `/content/we-retail/us/en/community/configuration` or `/content/we-retail/us/en/community/configuration/functions`.

As such:
- There is currently no way to exclude a non-page path - folders are coming along for the ride, whether desired or not
- If an exclusion pattern matches a folder path, the package creation errors out w/o any notification to the user.

Reply:
Addressed in the latest commit.